src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToGenerateSupportedTargetFrameworkAlias.cs

```diff
@@ -0,0 +1,156 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.IO;
+using FluentAssertions;
+using Microsoft.NET.TestFramework;
+using Microsoft.NET.TestFramework.Commands;
+using Xunit;
+using Xunit.Abstractions;
+using Microsoft.NET.TestFramework.Assertions;
+using Microsoft.NET.TestFramework.ProjectConstruction;
+using System.Xml.Linq;
+using System.Linq;
+using System.Collections.Generic;
+using NuGet.Frameworks;
+
+namespace Microsoft.NET.Build.Tests
+{
+    public class GivenThatWeWantToGenerateSupportedTargetFrameworkAlias : SdkTest
+    {
+        public GivenThatWeWantToGenerateSupportedTargetFrameworkAlias(ITestOutputHelper log) : base(log)
+        {}
+
+        [Theory]
+        [InlineData("netcoreapp3.1")]
+        [InlineData("net5.0")]
+        [InlineData("net6.0")]
+        [InlineData("netstandard2.1")]
+        [InlineData("net48")]
+        public void It_generates_supported_target_framework_alias_items(string currentTargetFramework)
+        {
+            TestTargetFrameworkAlias(currentTargetFramework, propertySetToTrue: null, new[]
+                {
+                    "netcoreapp3.0",
+                    "netcoreapp3.1",
+                    "net5.0",
+                    "net6.0",
+                    "netstandard2.0",
+                    "netstandard2.1",
+                    "net471",
+                    "net48"
+                });
+        }
+
+        [WindowsOnlyTheory]
+        [InlineData("net5.0-windows")]
+        [InlineData("net6.0-windows")]
+        public void It_generates_supported_target_framework_alias_items_when_targeting_windows(string currentTargetFramework)
+        {
+            TestTargetFrameworkAlias(currentTargetFramework, propertySetToTrue: null, new[]
+                {
+                    "netcoreapp3.0",
+                    "netcoreapp3.1",
+                    "net5.0-windows7.0",
+                    "net6.0-windows7.0",
+                    "netstandard2.0",
+                    "netstandard2.1",
+                    "net471",
+                    "net48"
+                });
+        }
+
+        [WindowsOnlyTheory]
+        [InlineData("net5.0", "UseWpf")]
+        [InlineData("net5.0", "UseWindowsForms")]
+        [InlineData("net5.0-windows", "UseWpf")]
+        [InlineData("net5.0-windows", "UseWindowsForms")]
+        [InlineData("netcoreapp3.1", "UseWpf")]
+        [InlineData("netcoreapp3.1", "UseWindowsForms")]
```

Comment:
Is it ok to use task level tests? These tests are expensive (several seconds for tests that starts from a project on disk). And we add about 30 minutes every release, it is no negligible

Reply:
Sure, just converted to unit tests. Look okay to you (and @dsplaisted)?

Reply:
We get a bit less coverage if we only do unit tests.  Mainly it doesn't cover how we set the default TargetPlatformVersion.

Reply:
We are able to test that the target platform version from the target platform moniker gets included in the target framework in [this test](https://github.com/dotnet/sdk/pull/13545/files/69c2724cdf27f4ecb484814744ca4ca17e9cedb2#diff-23f23314e105085486eb13f5b1650dccR54), is there more coverage you think we need?